enhancements/update/versioning-cincinnati-api.md

```diff
@@ -0,0 +1,116 @@
+The move to supplying requested version in media type field will require us to move towards a generic un-versioned Graph
+URI, with the move, we'll still support old URI for foreseeable future. 
+
+## Motivation
+
+### Goals
+1. Embed version information in json schema.
+2. Implement media type versioning for clients requesting a specific schema version of cincinnati
+3. Move to a generic un-versioned URI for cincinnati
+4. Cincinnati continues to support older clients after adding support for new clients for foreseeable future.
+5. Add metrics to identify hits on URIs and versions.
+
+### Non-Goals
+
+## Proposal
+
+### Implementation
+
+#### Version information in json schema 
+Embed the version information in json schema. The updated json schema should look like:
+```json
+{
+    "version": 1,
+    "nodes": [],
+    "edges": []
+}
+```
+
+#### Media Type Versioning 
+Currently, the versioning is done by using the versioned URI and does not support media type versioning. The user should 
+be able to request a specific version of cincinnati graph by supplying the version in media type field of the request. 
+for e.g. `ContentType: application/vnd.redhat.cincinnati.v1+json`. Here are some of the 
+[examples of media type versioning](https://github.com/opencontainers/image-spec/blob/v1.0.1/media-types.md).       
+In the event where the media type is not provided, or `application/json` is provided, cincinnati should return a default 
+response that is compatible with earlier versions (v1) so as not to break old clusters from upgrading. 
+
+
+#### URI Change
+The current URI follows `{URL}/v1/graph`. The new un-versioned URI will follow `{URL}/graph`. The existing URI can be 
+redirected to the new URI. This can be done through cincinnati. In the case where an old URI is being called, cincinnati 
+will by default return the graph that is compatible with `v1`.      
+
+
+#### Version metrics
+Cincinnati should also implement a way to understand how many clients are consuming the various versions. 
+This can be done by implementing metrics for old URI vs new URI. Similarly, metrics should be implemented to understand
+the consumption of various media types (versions). This will help us to make decisions on deprecating the old versions 
+and URI. 
+
+
+### Risks and Mitigations
+* As there are some customers that have old URI hardcoded while creating the cluster, we will have to support old URI 
+  for indefinite period of time. We can redirect the old URI to new one to avoid maintaining different URIs, but we'll 
+  still have to maintain different versions and v1 for indefinite period of time. 
+
+
+## Design Details
+
+### Test Plan
+* cincinnati tests to make sure we are not breaking older versions 
+* cincinnati returns correct resource version when the content type is provided.
+
+### Graduation Criteria
+GA. When it works, we ship it.
+
+
+### Version Skew Strategy
+No special consideration.
```

Comment:
we may want to point out that the need to support v1 forever only applies to the URI.  We control all the supported clients, and [set maintenance cutoffs][1], so there will come a time when we no longer care about supporting clients who only speak v1.  The endless maintenance burden is just for clients that can speak v2, etc., but still make their v2 Cincinnati requests via the original, nominally v1, URI because we used to (before openshift/installer#4112, 4.7), stash that in `spec.upstream`, and robots touching `spec` properties to try to auto-recover that for born-before-4.7 clusters by clearing `spec.upstream` when it matches our old default would potentially override an admin who explicitly wanted the cluster pulling from that URI.  And maybe they were on a restricted network with a DNS horizon or some such, and they really meant to stick with that upstream URI?  Anyway, some words about "we are stuck supporting the current URI forever, but do not have to support the v1 payload format forever" would be good.

[1]: https://access.redhat.com/support/policy/updates/openshift/#dates